### DISCUSSION TITLE
PWM not linear on WT32-ETH01

### DISCUSSION THREAD

**Discussion by MrJoki007** [arendst/Tasmota#19305] [Q&A]
Hello,
i want to use a WT32-ETH01 to output a pwm signal.
The problem i have is when i set the slider to around 50% i get a duty cycle of around 25%.
The only correct values are when the slider is set to 0% or 100%.

Ive tried two different WT32-ETH01 boards and different output pins.

![IMG_7539](https://github.com/arendst/Tasmota/assets/8217564/33664bb9-48e9-45d9-9609-d75c97939c7c)
![IMG_7536](https://github.com/arendst/Tasmota/assets/8217564/867d7eff-66c3-4708-b97a-0042b8d47554)
![IMG_7537](https://github.com/arendst/Tasmota/assets/8217564/952d0112-be2e-4d99-99a8-61ed35be6d26)
![IMG_7538](https://github.com/arendst/Tasmota/assets/8217564/d6446f5b-d656-4bec-812c-49d3eaa30038)

**Comment by sfromis** [comment]
The default config for PWM is to see it as a light, where it is highly beneficial to have the gamma correction you see. For other usage, you can turn off gamma correction with the command LedTable. However, if you want "full control" over the PWM, you may want to consider dropping the light-oriented Tasmota support (0-100), to instead use SetOption15 to switch to basic PWM control, with the PWM command to select duty cycle in 1024 steps.
